lang/compiler.go

```diff
@@ -54,11 +54,15 @@ func (c SpecCompiler) CompilerType() flux.CompilerType {
 
 // ASTCompiler implements Compiler by producing a Spec from an AST.
 type ASTCompiler struct {
+	Ext *ast.File    `json:"extern"`
 	AST *ast.Package `json:"ast"`
 	Now func() time.Time
 }
 
 func (c ASTCompiler) Compile(ctx context.Context) (*flux.Spec, error) {
+	if c.Ext != nil {
+		c.AST.Files = append([]*ast.File{c.Ext}, c.AST.Files...)
```

Comment:
This seems like the wrong design. Everytime `Compile` is called the file will be prepended to the package. We should not be modifying the AST from within the compiler.